packages/app-desktop/gui/NoteEditor/utils/clipboardUtils.test.ts

```diff
@@ -29,6 +28,11 @@ describe('clipboardUtils', () => {
 		}
 	});
 
+	// We are mocking console.error for the following tests
+	// since jsdom throws errors to console when we try to load an invalid img
+	// https://github.com/facebook/jest/pull/5267#issuecomment-356605468
+	console.error = jest.fn();
```

Comment:
Could you call that in beforeEach please. Then restore in afterEach.

Reply:
Actually `afterEach` will break the mocking. The thing is those functions that use dom elements are not async, i.e they don't wait for the images to load by jsdom, thus what happens is the console.error is thrown after the test has ended and if we clean up the mock, the logs will be visible.

- One way can be to use a `describe` block to fence out those tests that need the mocking and mock console.error only for that block.
- Another way may be to create a custom test environment in jest, although I'm not very sure if it can satisfy our needs, need to play around it a bit. Documentations are bit scarce. https://jestjs.io/docs/configuration#testenvironment-string

Reply:
Instead of mocking errors in the other tests, I have created a new custom test environment for jest that will load remote resources instead of changing the global settings in `jest.config.js`. I think this is a better approach because:

- We are not changing any global settings for jest, so that all previous tests run as they used to.
- We are adding the new functionality of loading resources as a separate module that can be loaded by tests if they need it.

Reply:
That makes a lot of sense and indeed a much better approach, thanks for adding this.